Add docs for Connect config (#22898)
gravitational/teleport: docs/pages/connect-your-client/teleport-connect.mdx

</TabItem>
</Tabs>

> ## Configuration

Reviewer: ```suggestion
## Configuration

```

| `keymap.openClusters`         | `Command+E` on macOS<br/>`Ctrl+E` on Windows/Linux                                                                   | Shortcut to open the cluster selector.                                 |
| `keymap.openProfiles`         | `Command+I` on macOS<br/>`Ctrl+I` on Windows/Linux                                                                   | Shortcut to open the profile selector.                                 |
| `keymap.openCommandBar`       | `Command+K` on macOS<br/>`Ctrl+K` on Windows/Linux                                                                   | Shortcut to open the command bar.                                      |
| `terminal.fontFamily`         | `Menlo, Monaco, monospace` on macOS<br/>`Consolas, monospace` on Windows<br/>`'Droid Sans Mono', monospace` on Linux | Font family for the terminal.                                          |
| `terminal.fontSize`           | 15                                                                                                                   | Font size for the terminal.                                            |

<Admonition
  type="note"
  title="Note"
>

>   The additional `$schema` property present in the config file allows text editors to provide autocompletion.

Reviewer: This is the only time we mention the `$schema` property, so I think it would be good to include an example of how to use it and when it would be useful.

Author: Technically, it should work out of the box, but I can add something like this: 
`It is a JSON schema file that describes the config properties. Your editor should recognize it automatically.`.

WDYT?

Reviewer: Just to clarify, when you install Teleport Connect, does it write a config file automatically? And does that config file include the `$schema` property? 

If so, we might want to clarify that users don't need to create their own config file, that the `$schema` property exists in the default config file, and that users should not change it.

Author: >when you install Teleport Connect, does it write a config file automatically? And does that config file include the $schema property?

Yes and yes.

>If so, we might want to clarify that users don't need to create their own config file

The paragraph starts with:
```
Teleport Connect can be configured by editing the `app_config.json` file.
To open it, click the More Options icon `⋮` -> Open Config File.
```
IMO it is clear that the file has been already created, isn't it? 

> ...that the $schema property exists in the default config file, and that users should not change this property.

I added a line saying that it should not be modified.

</Admonition>

> ### Configuring keyboard shortcuts

Reviewer: ```suggestion
### Configuring keyboard shortcuts

```
We're standardizing on one line break after each heading

</TabItem>
</Tabs>

## Configuration

> Teleport Connect can be configured by editing the `app_config.json` file.

Reviewer: ```suggestion
Teleport Connect can be configured by editing the `app_config.json` file, which it creates on first installation.
```

Author: Changed to `first launch` to be more precise.

| `keymap.openProfiles`         | `Command+I` on macOS<br/>`Ctrl+I` on Windows/Linux                                                                   | Shortcut to open the profile selector.                                 |
| `keymap.openCommandBar`       | `Command+K` on macOS<br/>`Ctrl+K` on Windows/Linux                                                                   | Shortcut to open the command bar.                                      |
| `terminal.fontFamily`         | `Menlo, Monaco, monospace` on macOS<br/>`Consolas, monospace` on Windows<br/>`'Droid Sans Mono', monospace` on Linux | Font family for the terminal.                                          |
| `terminal.fontSize`           | 15                                                                                                                   | Font size for the terminal.                                            |

<Admonition
  type="note"
  title="Note"
>
  The additional `$schema` property present in the config file allows text editors to provide autocompletion.

>   It should not be modified.

Reviewer: ```suggestion
  Users must not modify this property.
```

Author: TBH I would leave it as is.

`Must not modify` sounds like the app would stop working if the user changed the property (in fact, it only affects autocompletion). Also, we override this property with the correct path on every launch, so even if you change this, we will bring back the original value.

</TabItem>
</Tabs>

## Configuration
Teleport Connect can be configured by editing `app_config.json` file.

> To open it, click More Options icon `⋮` -> Open Config File.

Reviewer 2: ```suggestion
To open it, click the More Options icon `⋮` -> Open Config File.
```

| `terminal.fontFamily`         | `Menlo, Monaco, 'Courier New', monospace` on macOS<br/>`'Consolas', 'Courier New', monospace` on Windows<br/>`'Droid Sans Mono', 'Courier New', monospace` on Linux | Font family for the terminal.                                          |
| `terminal.fontSize`           | 15                                                                                                                                                                  | Font size for the terminal.                                            |

<Admonition
  type="note"
  title="Note"
>
  In the config file there is an additional `$schema` property. It is a link to the JSON schema that
  describes the available config options.
  It allows editors to provide autocompletion.
</Admonition>

### Configuring keyboard shortcuts
A valid shortcut contains at least one modifier and a single key code, for example `Shift+Tab`.
Function keys (`F21`-`F24`) do not require a modifier.
Modifiers and a key code must be combined by `+` character.

> <Tabs>
>   <TabItem label="macOS">
>     Available platform modifiers: `Control`, `Option`, `Shift`, `Command`.
>   </TabItem>
>   <TabItem label="Windows/Linux">
>     Available platform modifiers: `Ctrl`, `Alt`, `Shift`.
>   </TabItem>
> </Tabs>

Reviewer 3: ```suggestion
Available modifiers:

* `Control`, `Option`, `Shift`, `Command` on macOS.
* `Ctrl`, `Alt`, `Shift` on Windows and Linux.
```

`<Tabs>` here IMHO obscures too much information which can be shown in a simpler way without this component.

Available key codes:
- `0` to `9`
- `A` to `Z`
- `F1` to `F24`
- `,`, `.`, `/`, `\`, `` ` ``, `-`, `=`, `;`, `'`, `[`, `]`
- `Space`, `Tab`, `CapsLock`, `NumLock`, `ScrollLock`, `Backspace`, `Delete`, `Insert`, `Enter`, `Up`, `Down`, `Left`, `Right`, `Home`, `End`, `PageUp`, `PageDown`, `Escape`, `IntlBackslash`

## Telemetry

(!docs/pages/includes/teleport-connect-telemetry.mdx!)

### Disabling telemetry

> If you initially agreed to share telemetry data, but now want to opt out, you need to update the `app_config.json` file (see [Configuration](#configuration)):

Reviewer 3: ```suggestion
If you initially agreed to share telemetry data, but now want to opt out, you need to set `usageReporting.enabled` in the config to `false` (see [Configuration](#configuration)):
```
